src/toniarts/openkeeper/world/TileData.java

```diff
@@ -216,7 +216,7 @@ public boolean applyDamage(int damage) {
      * @return true if the tile is at max
      */
     public boolean applyHealing(int healing) {
-        health = Math.min(getTerrain().getMaxHealth(), health + healing);
+        health = (int) Math.min(getTerrain().getMaxHealth(), (long) health + healing);

```

Comment:
after claiming the portal we `applyHealing(Integer.MAX_VALUE);`. What happen if we make `(int) 10 + Integer.MAX_VALUE` ? Ea -2431254325 (some big negative value).

`Math.min(getTerrain().getMaxHealth(), health + healing);` return this negative value without cast.